File: typescript/ltx/src/formatting.ts
/**
 * @interplanet/ltx — Formatting utilities
 */

function _pad(n: number): string {
  return String(Math.floor(n)).padStart(2, '0');
}

/**
 * Format a duration in seconds as `HH:MM:SS` or `MM:SS`.
 */
export function formatHMS(sec: number): string {
  if (sec < 0) sec = 0;
  const h = Math.floor(sec / 3600);
  const m = Math.floor((sec % 3600) / 60);
  const s = Math.floor(sec % 60);
  if (h > 0) return `${_pad(h)}:${_pad(m)}:${_pad(s)}`;
  return `${_pad(m)}:${_pad(s)}`;
}

/**
 * Format a Date, timestamp (ms), or ISO string as `HH:MM:SS UTC`.
 */
export function formatUTC(dt: Date | number | string): string {
  return new Date(dt as number).toISOString().slice(11, 19) + ' UTC';
}

File: python/ltx/src/interplanet_ltx/_formatting.py
"""Formatting utilities for LTX — Python port of ltx-sdk.js formatHMS/formatUTC."""

from datetime import datetime, timezone


def format_hms(sec: float) -> str:
    """Format seconds as HH:MM:SS or MM:SS."""
    if sec < 0:
        sec = 0
    sec = int(sec)
    h = sec // 3600
    m = (sec % 3600) // 60
    s = sec % 60
    if h > 0:
        return f'{h:02d}:{m:02d}:{s:02d}'
    return f'{m:02d}:{s:02d}'


def format_utc(dt) -> str:
    """Format a datetime or UTC millisecond timestamp as 'HH:MM:SS UTC'."""
    if isinstance(dt, (int, float)):
        dt = datetime.fromtimestamp(dt / 1000, tz=timezone.utc)
    elif isinstance(dt, str):
        dt = datetime.fromisoformat(dt.replace('Z', '+00:00'))
    return dt.strftime('%H:%M:%S') + ' UTC'
